### PR TITLE
[PoC] Low Latency Participant [DPP-784]

### PR DESCRIPTION
**Pull request by tudor-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/Config.scala [hunk 1]
```diff
@@ -85,7 +85,7 @@ object Config {
       acsIdQueueLimit = IndexConfiguration.DefaultAcsIdQueueLimit,
       configurationLoadTimeout = Duration.ofSeconds(10),
       commandConfig = CommandConfiguration.default,
-      enableInMemoryFanOutForLedgerApi = false,
+      enableInMemoryFanOutForLedgerApi = true,
```
Comment (tudor-da) [inline ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/Config.scala:88]: In-memory fan-out enabled by default

File: ledger/sandbox-on-x/src/classic/scala/com/daml/platform/ledger/ConfigConverter.scala [hunk 2]
```diff
@@ -66,7 +66,7 @@ object ConfigConverter {
       acsIdQueueLimit = sandboxConfig.acsIdQueueLimit,
       configurationLoadTimeout = sandboxConfig.configurationLoadTimeout,
       commandConfig = sandboxConfig.commandConfig,
-      enableInMemoryFanOutForLedgerApi = false,
+      enableInMemoryFanOutForLedgerApi = true,
```
Comment (tudor-da) [inline ledger/sandbox-on-x/src/classic/scala/com/daml/platform/ledger/ConfigConverter.scala:69]: Enabled for all

File: ledger/participant-integration-api/src/main/scala/platform/index/BuffersUpdater.scala [hunk 3]
```diff
@@ -56,9 +61,15 @@ private[index] class BuffersUpdater(
           randomFactor = 0.0,
         )
       )(() => subscribeToTransactionLogUpdates(updaterIndex.get))
-      .map { case ((offset, eventSequentialId), update) =>
-        updateCaches(offset, update)
-        updaterIndex.set(Some(offset -> eventSequentialId))
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/index/BuffersUpdater.scala:61]: The updaterIndex is no longer set anywhere. This means that re-subscription to the upstream source is no longer possible

File: ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/InitializeParallelIngestion.scala [hunk 4]
```diff
@@ -59,10 +61,20 @@ private[platform] case class InitializeParallelIngestion(
     } yield InitializeParallelIngestion.Initialized(
       initialEventSeqId = ledgerEnd.lastEventSeqId,
       initialStringInterningId = ledgerEnd.lastStringInterningId,
-      readServiceSource = readService.stateUpdates(beginAfter = ledgerEnd.lastOffsetOption),
+      readServiceSource = readService
+        .stateUpdates(beginAfter = ledgerEnd.lastOffsetOption)
+        .alsoTo(
+          com.daml.platform.index.LedgerBuffersUpdater
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/InitializeParallelIngestion.scala:67]: That's a bit of a mouthful

File: ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/InitializeParallelIngestion.scala [hunk 5]
```diff
@@ -59,10 +61,20 @@ private[platform] case class InitializeParallelIngestion(
...
+            .flow(ledgerEnd.lastEventSeqId)
+            .to(Sink.foreachAsync[((Offset, Long), TransactionLogUpdate)](1) { o =>
+              buffersUpdatesQueue.offer(o).map {
+                case QueueOfferResult.Enqueued =>
+                case r => throw new RuntimeException(s"Offset/update pair ($o) not enqueued: $r")
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/InitializeParallelIngestion.scala:72]: That's hell a lot of details printed into the exception

File: ledger/participant-integration-api/src/main/scala/platform/index/BuffersUpdater.scala (hunk 3 above)
Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/index/BuffersUpdater.scala:61]: Indeed. It was a conscious decision to leave this detail out in the first iteration.

Comment (mziolekda) [reply ledger/participant-integration-api/src/main/scala/platform/index/BuffersUpdater.scala:61]: You should leave a TODO then, it is easily overlooked

Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/index/BuffersUpdater.scala:61]: Good point.

**Comment by tudor-da** [comment]
Replaced by https://github.com/digital-asset/daml/pull/13537

**Pull request closed without merging**